src/main/dialogs/captureToolDialog.js

```diff
@@ -0,0 +1,294 @@
+        message: "Select 'Run Capture' to run the capture diagnostics and store them to disk.\n\n  After capture they will be stored in:\n" + captureDirName
+    });
+
+    if (dailogPromise === 1) {
+        fs.mkdirSync(captureDirName);
+        gpii.app.captureTool.generateDiagnostics(captureToolUtils).then(
+            function (data) {
+                // 0. Solutions
+                fs.appendFileSync(captureDirName + "/solutions.json", JSON.stringify(data.solutions, null, 4));
+                // 1. Installed Solutions
+                fs.appendFileSync(captureDirName + "/installedSolutions.json", JSON.stringify(data.installedSolutions, null, 4));
+                // 2. Full System Settings Capture
+                fs.appendFileSync(captureDirName + "/settingsCapture.json", JSON.stringify(data.settingsCapture, null, 4));
+
+                electron.dialog.showMessageBox({
+                    message: "Capture diagnostics successfully taken and are now stored in this directory:\n" + captureDirName
+                });
+            },
+            function (err) {
+                electron.dialog.showMessageBox({
+                    message: "There was a problem capturing the diagnostics:\n" + err
+                });
+                fluid.log("Error running capture diagnostics. ", err);
+            }
+        );
+    }
+};
+
+/*
+ * Generates and returns a set of diagnostics for the current local machine.
+ *
+ * Returns a promise containing these.
+ */
+gpii.app.captureTool.generateDiagnostics = function (captureToolUtils) {
+    return fluid.promise.fireTransformEvent(captureToolUtils.events.onCollectDiagnostics, {});
+};
+
+gpii.app.captureTool.init = function (that, flowManager) {
+    ipcMain.on("getInstalledSolutions", function (event /*, arg */) {
+        flowManager.capture.getInstalledSolutions().then(
+            function (data) {
+                event.sender.send("sendingInstalledSolutions", data);
+                var runningSolutions = {};
+                fluid.each(data, function (solutionEntry, solutionId) {
+                    try {
+                        fluid.log("ABOUT TO CHECK PROCESS FOR: ", solutionId);
+                        var isRunningState = gpii.processReporter.handleIsRunning(solutionEntry);
+                        fluid.log("RUNNING STATE: ", isRunningState);
+                        if (isRunningState) {
+                            runningSolutions[solutionId] = solutionEntry;
+                        }
+                    } catch (err) {
+                        fluid.log("Exception trying to look up", solutionEntry.isRunning);
+                    }
+                });
+                event.sender.send("sendingRunningSolutions", runningSolutions);
+            },
+            function (err) {
+                event.sender.send("sendingInstalledSolutions", {isError: true, message: err});
+            }
+        );
+    });
+
+    ipcMain.on("getAllSolutionsCapture", function (event, arg) {
```

Comment:
Rather than manually bash on ipcMain like this in a private function, you can use the "channelUtils" that the Astea team put together as at least a half-assed kind of declarative process bridge - https://github.com/GPII/gpii-app/blob/master/src/shared/channelUtils.js#L28

Reply:
Removed all bashing on of icpMain and icpRenderer